Spanner/src/RequestHeaderTrait.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Copyright 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner;
+
+use Google\Cloud\Spanner\Session\SessionPoolInterface;
+
+/**
+ * Shared methods for request headers.
+ */
+trait RequestHeaderTrait
+{
+    /**
+     * Add the `x-goog-spanner-route-to-leader` header value to the request.
+     *
+     * @param array $args
+     * @param bool $value
+     * @param string $context
+     * @return array
+     */
+    private function addLarHeader(
+        array $args,
+        bool $value = true,
+        string $context = SessionPoolInterface::CONTEXT_READWRITE
+    ) {
+        // If value is false, unset LAR header.
+        if (!$value) {
+            return $this->conditionallyUnsetLarHeader($args, $value);
```

Comment:
So, if we are setting the header in this function itself, then why do we need to unset it here?
Shouldn't we just `return $args`? Are there other places where we are setting this header?

Reply:
Its like this, lets say

Reply:
I have considered a situation where the user send an argument `$args['headers']['x-goog-spanner-route-to-leader'] = ['true']` at function level for example `public commit` function in `Grpc.php`. Then, when `addLarHeader` function gets invoked from the `Grpc::commit` function. At that time, we need to override this with the client level `larEnabled`. Thats why I added this logic here. Hope its clear now.

Reply:
Sounds good